### PR TITLE
Gravity

### PR DESCRIPTION
**Pull request by chai-amy**
- small json creation tool
- upload scripts for vali to chain
- retrieval scripts for valis and miners
- test file for converting from json to DataDesirabilityLookup

### CODE REVIEW

File: dynamic_desirability/total.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[
```
[the remark this answers is not in the document]
Comment (chai-amy) [reply dynamic_desirability/total.json:1]: We don't, I'll delete

File: dynamic_desirability/my_preferences.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[
```
[the remark this answers is not in the document]
Comment (chai-amy) [reply dynamic_desirability/my_preferences.json:1]: We don't technically need a prebuilt my_preferences file, but if validators want to try running it out of the box then this way there won't be any errors thrown. I've replaced this with a valid submission with labels named some variant of "example".

File: rewards/data.py [hunk 3]
```diff
@@ -35,7 +35,8 @@ def validate_label_scale_factors(
     ) -> Dict[str, float]:
         """Validates the label_scale_factors field."""
         for label, scale_factor in value.items():
-            if scale_factor < -1.0 or scale_factor > 1.0:
+            # Max label weight for one active validator putting 100% on one label = 23.333
+            if scale_factor < -1.0 or scale_factor > 23.4:
```
[the remark this answers is not in the document]
Comment (chai-amy) [reply rewards/data.py:39]: Max validator weight with 1 active validator putting 100% on one label at 70/30 validator/subnet weight is 23.3333

File: dynamic_desirability/desirability_retrieval.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+from dynamic_desirability.chain_utils import ChainPreferenceStore
+from common import constants
+from common.data import DataLabel, DataSource
+from rewards.data import DataSourceDesirability, DataDesirabilityLookup
+from dynamic_desirability.constants import NETWORK, NETUID, REPO_URL, PREFERENCES_FOLDER, DEFAULT_JSON_PATH, AGGREGATE_JSON_PATH
+from dynamic_desirability.constants import TOTAL_VALI_WEIGHT, REDDIT_SOURCE_WEIGHT, X_SOURCE_WEIGHT
+from dynamic_desirability.gravity_config import WALLET_NAME, HOTKEY_NAME
+
+
+def get_validator_data(metagraph: bt.metagraph) -> Dict[str, Dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (chai-amy) [reply dynamic_desirability/desirability_retrieval.py:19]: I will move the function in common/utils.py

File: dynamic_desirability/desirability_retrieval.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        data = json.load(file)
+    
+    distribution = {}
+    
+    for source in data:
+        source_name = source['source_name']
+        label_weights = source['label_weights']
+        
+        # Set predefined weights
+        weight = REDDIT_SOURCE_WEIGHT if source_name == "reddit" else X_SOURCE_WEIGHT
```
[the remark this answers is not in the document]
Comment (chai-amy) [reply dynamic_desirability/desirability_retrieval.py:146]: I will add *weight* into DataSource

**Review by Arrmlet — COMMENTED** [review]
reviewed

**Review by Arrmlet — APPROVED** [review]
reviewed

**Review by Arrmlet — APPROVED** [review]
I reviewed

**Review by bkb2135 — APPROVED** [review]

**Pull request merged**